src/main/java/school/lemon/changerequest/java/introduction/hw2/Task9.java

```diff
@@ -10,19 +10,18 @@
         }
 
 
-        String shema = "([a-z]+(?=\\:\\/\\/))";  // for all leters before ://
-        String host = "(([a-z0-9.-_])+ (?=\\:))";  // for all leters , numbers and . -_ before :  (but if url dont have :port ?)
-        String port = "(?=\\:[0-9]+)";  // for all numbers after :
-        Pattern pshema = Pattern.compile(shema);
-        Pattern phost = Pattern.compile(host);
-        Pattern pport = Pattern.compile(port);
-        Matcher mshema = pshema.matcher(URL);
-        Matcher mhost = phost.matcher(URL);
-        Matcher mport = pport.matcher(URL);
 
+String url =" ( (?<=\\:\\/\\/) [\\w] +.) ? ((?<= \\/\\/) [\\w\\d.-_] +.) ((?<=\\: ) [\\d] ) ";
```

Comment:
I'm pretty sure, that your regexp is not correct.
Schema could contain only [a-zA-Z0-9].
Host could contain [a-zA-Z0-9-_.]
Port could have more than 1 number